src/Collection.js

```diff
@@ -264,7 +264,7 @@ export default Component.extend({
         // Here, the currently visted record does not correspond to any record
         // in the collection, so (optionally) instantiate this record and insert
         // it into the collection
-        record = this.mapper ? this.mapper.createRecord(record, opts) : record
+        record = this.mapper && !this.mapper.is(record) ? this.mapper.createRecord(record, opts) : record
```

Comment:
According to my debugging record is always here an instance of `mapper` and if not then it should be converted through `createRecord`. JSData supports only 3 ways of model creation: `createRecord`, `createInstance`, `wrap` (all goes through `createRecord`). All others should be unacceptable. If people want to instantiate model using `new` it shouldn't work properly and this is what should be noted in docs.

Please correct me if I'm wrong, so then I will change back this